scheduler/scheduler_test.go

```diff
@@ -384,3 +389,83 @@ func testInboundContentType(node interface{}) {
 		So(t.InboundContentType, ShouldNotEqual, "")
 	}
 }
+
+func TestSubscribeAfterNewPluginLoaded(t *testing.T) {
+	// Create workflow map for task
+	wf := new(wmap.WorkflowMap)
+	cn := wmap.NewCollectWorkflowMapNode()
+	cn.Config["/intel/mock/foo"] = make(map[string]interface{})
+	cn.Config["/intel/mock/foo"]["password"] = "mockpass"
+	cn.Config["/intel/anothermock/foo"] = make(map[string]interface{})
+	cn.Config["/intel/anothermock/foo"]["password"] = "anothermockpass"
+	cn.AddMetric("/intel/*/foo", 1)
+	wf.CollectNode = cn
+
+	// Create plugin manager
+	c := control.New(control.GetDefaultConfig())
+	c.Start()
+	time.Sleep(100 * time.Millisecond)
+
+	var schTask *task
+	var s *scheduler
+	mock1Path := path.Join(PluginPath, "snap-collector-mock1")
+	anotherMock1Path := path.Join(PluginPath, "snap-collector-anothermock1")
+
+	Convey("RefreshTaskSubscriptions()", t, func() {
+		Convey("Create new scheduler and set metric manager", func() {
+			s = New(GetDefaultConfig())
+			s.SetMetricManager(c)
+			err := s.Start()
+			So(err, ShouldBeNil)
+			time.Sleep(100 * time.Millisecond)
```

Comment:
It appears that all of the sleeps in this test can be removed if you register and use an event handler.